src/node_quic_session.cc

```diff
@@ -1527,7 +1527,9 @@ void QuicSession::StreamReset(
 
 void QuicSession::UpdateIdleTimer() {
   CHECK_NOT_NULL(idle_);
-  uint64_t timeout = ngtcp2_conn_get_idle_expiry(Connection()) / 1000000UL;
+  uint64_t timeout = (ngtcp2_conn_get_idle_expiry(Connection()) - uv_hrtime()) / 1000000UL;
+  if (timeout < 0)
+    timeout = 0;
```

Comment:
`timeout < 0` never happens because it is unsigned; and `0` won’t schedule the time afaik, whereas `1` will…

Should this maybe overall look similar to the retransmit timer code as in e.g. https://github.com/nodejs/quic/pull/157/files?

Reply:
That's right.. I would like to follow the https://github.com/nodejs/quic/pull/157/files